dftimewolf/lib/processors/turbinia_gcp_tam.py

```diff
@@ -0,0 +1,361 @@
+    run_all_jobs (bool): Whether to run all jobs or to remove slow-running jobs:
+        'StringsJob', 'BinaryExtractorJob', 'BulkExtractorJob', 'PhotorecJob'.
+    sketch_id (int): The Timesketch sketch id
+    turbinia_region (str): GCP region in which the Turbinia server is running.
+    turbinia_zone (str): GCP zone in which the Turbinia server is running.
+  """
+
+  def __init__(self,
+               state: "state.DFTimewolfState",
+               name: Optional[str]=None,
+               critical: bool=False) -> None:
+    """Initializes a Turbinia base processor.
+
+    Args:
+      state (state.DFTimewolfState): recipe state.
+      name (Optional[str]): The module's runtime name.
+      critical (Optional[bool]): True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(TurbiniaProcessorThreadedBase, self).__init__(
+        state, name=name, critical=critical)
+    self.turbinia_config_file = ''
+    self._output_path = str()
+    self.client = None  # type: turbinia_client.BaseTurbiniaClient
+    self.instance = None
+    self.project = str()
+    self.run_all_jobs = False
+    self.sketch_id = int()
+    self.turbinia_region = None
+    self.turbinia_zone = str()
+
+  def _DeterminePaths(
+      self,
+      task_data: List[Dict[str, Any]]) -> Tuple[List[str], List[str]]:
+    """Builds lists of local and remote paths from data retured by Turbinia.
+
+    This finds all .plaso, hashes.json, and BinaryExtractorTask files in the
+    Turbinia output, and determines if they are local or remote (it's possible
+    this will be running against a local instance of Turbinia).
+
+    Args:
+      task_data (list[dict]): List of dictionaries representing Turbinia task
+          data.
+
+    Returns:
+      tuple[list, list]: A tuple of two lists. The first element contains the
+          local paths, the second element contains the remote (GS) paths.
+    """
+    local_paths = []
+    gs_paths = []
+    for task in task_data:
+      # saved_paths may be set to None
+      saved_paths = task.get('saved_paths') or []
+      for path in saved_paths:
+
+        if path.endswith('.plaso') or \
+            path.endswith('BinaryExtractorTask.tar.gz') or \
+            path.endswith('hashes.json'):
+
+          if path.startswith('gs://'):
+            gs_paths.append(path)
+          else:
+            local_paths.append(path)
+
+    return local_paths, gs_paths
+
+  def _DownloadFilesFromGCS(self,
+                            timeline_label: str,
+                            gs_paths: List[str]) -> List[Tuple[str, str]]:
+    """Downloads files stored in Google Cloud Storage to the local filesystem.
+
+    Args:
+      timeline_label (str): Label to use to construct the path list.
+      gs_paths (List[str]):  gs:// URI to files that need to be downloaded
+          from GS.
+
+    Returns:
+      list:
```

Comment:
This may trigger some weird things on auto generated doc. Perhaps `list[tuple[str, str]]`: Tuple containing....